File: app/src/main/java/com/hiddenramblings/tagmo/charset/CharsetCompat.java
package com.hiddenramblings.tagmo.charset;

import android.os.Build;

import com.hiddenramblings.tagmo.eightbit.io.Debug;

import java.nio.charset.Charset;
import java.nio.charset.StandardCharsets;

@SuppressWarnings("all")
public class CharsetCompat {
    public static Charset UTF_8 = Debug.isNewer(Build.VERSION_CODES.KITKAT)
            ? UTF_8 = StandardCharsets.UTF_8
            : Charset.forName("UTF-8");
    public static Charset UTF_16 = Debug.isNewer(Build.VERSION_CODES.KITKAT)
            ? StandardCharsets.UTF_16
            : Charset.forName("UTF-16");
    public static Charset ISO_8859_1 = Debug.isNewer(Build.VERSION_CODES.KITKAT)
            ? StandardCharsets.ISO_8859_1
            : Charset.forName("ISO-8859-1");
    public static Charset US_ASCII = Debug.isNewer(Build.VERSION_CODES.KITKAT)
            ? StandardCharsets.US_ASCII
            : Charset.forName("US_ASCII");
    public static Charset UTF_16BE = Debug.isNewer(Build.VERSION_CODES.KITKAT)
            ? StandardCharsets.UTF_16BE
            : Charset.forName("UTF-16BE");
    public static Charset UTF_16LE = Debug.isNewer(Build.VERSION_CODES.KITKAT)
            ? StandardCharsets.UTF_16LE
            : Charset.forName("UTF-16LE");
}

File: app/src/main/java/com/hiddenramblings/tagmo/charset/CharsetCompat.kt
package com.hiddenramblings.tagmo.charset

import com.hiddenramblings.tagmo.eightbit.os.Version
import java.nio.charset.Charset
import java.nio.charset.StandardCharsets

object CharsetCompat {
    @JvmField
    val UTF_8: Charset =
        if (Version.isKitKat) StandardCharsets.UTF_8 else Charset.forName("UTF-8")
    @JvmField
    val UTF_16: Charset =
        if (Version.isKitKat) StandardCharsets.UTF_16 else Charset.forName("UTF-16")
    @JvmField
    val ISO_8859_1: Charset =
        if (Version.isKitKat) StandardCharsets.ISO_8859_1 else Charset.forName("ISO-8859-1")
    @JvmField
    val US_ASCII: Charset =
        if (Version.isKitKat) StandardCharsets.US_ASCII else Charset.defaultCharset() // Unknown
    @JvmField
    val UTF_16BE: Charset =
        if (Version.isKitKat) StandardCharsets.UTF_16BE else Charset.forName("UTF-16BE")
    @JvmField
    val UTF_16LE: Charset =
        if (Version.isKitKat) StandardCharsets.UTF_16LE else Charset.forName("UTF-16LE")
}